File: 11. Container With Most Water/Solution.go
package main

func maxArea(height []int) int {
    maxArea := 0
    left := 0
    right := len(height) - 1
    for left < right {
        minHeight := height[left]
        if height[right] < minHeight {
            minHeight = height[right]
        }
        area := (right - left) * minHeight
        if area > maxArea {
            maxArea = area
        }
        for height[left] <= minHeight && left < right {
            left++
        }
        for height[right] <= minHeight && left < right {
            right--
        }
    }
    return maxArea
}

	

File: 11. Container With Most Water/Solution.java
class Solution {
    public int maxArea(int[] height) {
        int maxArea = 0;
        int left = 0;
        int right = height.length - 1;
        while (left < right) {
            var minHeight = (height[left] < height[right])?height[left] : height[right];
            var area = (right - left) * minHeight;
            maxArea = (area > maxArea) ? area : maxArea;
            while (height[left] <= minHeight && left < right) {
                left++;
            }
            while (height[right] <= minHeight && left < right) {
                right--;
            }
        }
        return maxArea;       
    }
}